internal/infrastructure/elastic_search/query_device_features.go

```diff
@@ -0,0 +1,141 @@
+package elasticsearch
+
+import (
+	"fmt"
+
+	"github.com/pkg/errors"
+)
+
+type DeviceFeatures struct {
+	BatteryVoltage struct {
+		DocumentCount int64 `json:"doc_count"`
+	}
+	FuelPercentRemaining struct {
+		DocumentCount int64 `json:"doc_count"`
+	}
+	Odometer struct {
+		DocumentCount int64 `json:"doc_count"`
+	}
+	Oil struct {
+		DocumentCount int64 `json:"doc_count"`
+	}
+	Soc struct {
+		DocumentCount int64 `json:"doc_count"`
+	}
+	Speed struct {
+		DocumentCount int64 `json:"doc_count"`
+	}
+	Tires struct {
+		DocumentCount int64 `json:"doc_count"`
+	}
+}
+
+type DeviceFeaturesResp struct {
+	Aggregations struct {
```

Comment:
I had forgotten that `encoding/json` will match `"aggregations"` to `Aggregations`. That is nice.